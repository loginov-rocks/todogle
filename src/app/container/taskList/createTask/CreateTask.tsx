import { Button, TextField } from '@material-ui/core';
import * as React from 'react';

import useInput from '../../../../hooks/useInput';

import gapi from '../../../../services/gapi';
import { TaskResource } from '../../../../services/gapi/TaskResource';

interface Props {
  onCreate: (task: TaskResource) => void;
  taskListId: string;
}

const CreateTask = ({ onCreate, taskListId }: Props) => {
  const { value, bindProps, reset } = useInput('');

  const handleSubmit = (event: any) => {
    event.preventDefault();

    gapi.createTask(taskListId, { title: value })
      .then(task => {
        onCreate(task);
        reset();
      });
  };

  return (
    <form onSubmit={handleSubmit}>
      <TextField required {...bindProps} />
      <Button type="submit" variant="contained">Create Task</Button>
    </form>
  );
};

export default CreateTask;
