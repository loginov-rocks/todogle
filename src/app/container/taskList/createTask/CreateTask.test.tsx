import { shallow } from 'enzyme';
import * as React from 'react';

import CreateTask from './CreateTask';

it('matches snapshot', () => {
  const wrapper = shallow(
    <CreateTask onCreate={() => undefined} taskListId="task-list-id" />,
  );

  expect(wrapper).toMatchSnapshot();
});
