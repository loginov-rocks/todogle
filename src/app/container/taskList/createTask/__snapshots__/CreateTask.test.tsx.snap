// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<form
  onSubmit={[Function]}
>
  <WithStyles(ForwardRef(TextField))
    onChange={[Function]}
    required={true}
    value=""
  />
  <WithStyles(ForwardRef(Button))
    type="submit"
    variant="contained"
  >
    Create Task
  </WithStyles(ForwardRef(Button))>
</form>
`;
